lib/util/host-rules.ts

```diff
@@ -18,6 +18,16 @@ export interface HostRule {
 
 let hostRules: HostRule[] = [];
 
+export function findAll({ hostType }: { hostType: string }): HostRule[] {
+  return hostRules.filter(rule => rule.hostType === hostType);
+}
+
+export function getHost(hostRule: HostRule): string {
+  const url = URL.parse(hostRule.domainName);
+
+  return url.host || url.path;
```

Comment:
@rarkins I'm not sure I got the difference between `domainName` and `hostName` on the `HostRule` type. When is one or the other used ? Should we try and support both ?

Reply:
domainName is like `github.com`, hostName is like `api.github.com`. If Bundler's approach is *host* driven, then sometimes domainName might not work, but there may be no harm in trying it. `hostName` should be what you're looking for without needing any parsing.

Reply:
Do you think it would make sense to try `hostName || domainName` (parsing probably needed for `domainName` ? Also, are users expected to set the protocol when setting `hostName` ?

Reply:
Both hostName and domainName are protocol-free. It's probably ok to use the logic `hostName || domainName` as nothing to lose by doing so. One example is `github.com` (domainName) vs `api.github.com` (hostName)

Reply:
Made the change so that we ue `hostName || domainName`